bluehost-wordpress-plugin.php

```diff
@@ -63,6 +63,54 @@
 	require dirname( __FILE__ ) . '/bootstrap.php';
 }
 
+/**
+ * Check if platform is Jarvis
+ *
+ * @return bool
+ */
+function bh_is_jarvis() {
+	$is_jarvis = false;
+	$host      = array(
+		'dirs'      => explode( '/', ABSPATH ),
+		'user'      => get_current_user(),
+		'homedir'   => null,
+		'info_file' => null,
+	);
+
+	// Build host's home directory
+	foreach ( $host['dirs'] as $dir ) {
+		if ( ! empty( $dir ) ) {
+			$host['homedir'] = $host['homedir'] . '/' . $dir;
+
+			if ( $dir === $host['user'] ) {
+				break;
+			}
+		}
+	}
+
+	// Check for Jarvis .host-info file
+	if ( file_exists( $host['homedir'] . '/.host-info' ) ) {
+		$host['info_file'] = file_get_contents( $host['homedir'] . '/.host-info' );
+	}
+
+	// Check for Jarvis platform
+	if ( null !== $host['info_file'] && false !== strpos( $host['info_file'], 'platform = jarvis' ) ) {
+		$is_jarvis = true;
+	}
+
+	if (
+		null !== $host['info_file']
+		&& (
+			false !== stripos( $host['info_file'], 'platform = jarvis' )
+			|| false !== stripos( $host['info_file'], 'platform=jarvis' )
+		)
+	) {
+		$is_jarvis = true;
+	}
+
+	return $is_jarvis;
```

Comment:
I think you could simplify this a good bit by splitting on the current user value. Then take the first part and just tack on the current user and the `.host-info` filename. 

Then strip spaces, and use `stripos()` to account for any weird capitalization that might be there.

```suggestion
	$parts = explode( get_current_user(), ABSPATH );
	$file = $parts[0] . get_current_user() . '/.host-info';

	// Check for Jarvis .host-info file
	if ( file_exists( $file ) ) {
		$host_info = preg_replace( ' ', '', file_get_contents( $file ) );
		
		// Check for Jarvis platform
		if ( null !== $host_info && false !== stripos( $host_info, 'platform=jarvis' ) ) {
			return true;
		}
	}
	
	return false;
```